src/ml/neural_net/compute_context.hpp

```diff
@@ -26,6 +26,10 @@ class compute_context {
   /** Function that yields a compute context. */
   using factory = std::function<std::unique_ptr<compute_context>()>;
 
+  /** Function that yields a compute context (made to return
+   * tf_compute_context). */
+  using tf_factory = std::function<std::unique_ptr<compute_context>()>;
```

Comment:
You don't need a new name here. You can just use `factory` everywhere you were using `tf_factory` since they are both identical aliases for `std::function<std::unique_ptr<compute_context>()>`